keel-titus-plugin/src/main/kotlin/com/netflix/spinnaker/keel/titus/batch/ContainerJobConfig.kt

```diff
@@ -1,123 +1,112 @@
-          ),
-          "application" to application,
-          "containerAttributes" to containerAttributes,
-          "env" to environmentVariables,
-          "labels" to labels,
-          "resources" to mapOf(
-            "cpu" to cpus,
-            "disk" to diskMb,
-            "gpu" to gpus,
-            "memory" to memoryMb,
-            "networkMbps" to networkMbps,
-          ),
-          "retries" to retries,
-          "runtimeLimitSecs" to runtimeLimitSeconds,
-          "securityGroups" to securityGroups,
-          "iamProfile" to iamInstanceProfile,
-          "region" to region,
-          "capacityGroup" to capacityGroup,
-          "imageId" to imageId,
-          "entryPoint" to entrypoint
-        ),
-
-        "waitForCompletion" to waitForCompletion,
-        "cloudProvider" to cloudProvider,
-        "deferredInitialization" to deferredInitialization,
-        "credentials" to credentials,
-        "registry" to registry,
-        "account" to account,
-        "organization" to organization,
-        "repository" to repository,
-        when(type) {
-          TAG -> "tag"
-          DIGEST -> "digest"
-        } to versionIdentifier,
-
-        "region" to region,
-    )
+  val imageId: String = "$repository:${tag ?: digest}"
+  val cloudProvider: String = "titus"
+  val cloudProviderType: String = "aws"
 }
 
+/**
+ * Create a Spinnaker "Run Job" stage. A Run Job stage executes a container.
+ *
+ * See SubmitJobRequest class in clouddriver:
+ * https://github.com/spinnaker/clouddriver/blob/master/clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/client/model/SubmitJobRequest.java
+ */
+fun ContainerJobConfig.createRunJobStage() =
+  mutableMapOf(
+    "type" to RUN_JOB_TYPE,
+    "cloudProviderType" to cloudProviderType,
+    "cluster" to mapOf(
+      "capacity" to with(capacity) {
+        mapOf(
+          "min" to min,
+          "max" to max,
+          "desired" to desired
+        )
+      },
+      "application" to application,
+      "containerAttributes" to containerAttributes,
+      "env" to environmentVariables,
+      "labels" to labels,
+      "resources" to with(resources) {
+        mapOf(
+          "cpu" to cpu,
+          "disk" to disk,
+          "gpu" to gpu,
+          "memory" to memory,
+          "networkMbps" to networkMbps,
+        )
+      },
+      "retries" to retries,
+      "runtimeLimitSecs" to runtimeLimit.seconds,
+      "securityGroups" to securityGroups,
+      "iamProfile" to iamInstanceProfile,
+      "region" to location.region,
+      "capacityGroup" to capacityGroup,
+      "imageId" to imageId,
+      "entryPoint" to entrypoint
+    ),
+
+    "waitForCompletion" to waitForCompletion,
+    "cloudProvider" to cloudProvider,
+    "deferredInitialization" to deferredInitialization,
+    "credentials" to credentials,
+    "account" to location.account,
+    "region" to location.region,
+  ).also {
+    println(ObjectMapper().writeValueAsString(it))
```

Comment:
This println looks like a leftover from development.

Reply:
D'oh yes